contracts/src/main/kotlin/com/contractsAndStates/states/GameCurrencyStates.kt

```diff
@@ -1,13 +1,27 @@
 package com.contractsAndStates.states
 
-import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
 import com.r3.corda.lib.tokens.contracts.types.TokenType
-import net.corda.core.contracts.Amount
-import net.corda.core.identity.Party
-import net.corda.core.internal.uncheckedCast
 import net.corda.core.serialization.CordaSerializable
 import java.math.BigDecimal
 
+enum class ResourceType(val displayName: String) {
+    Wheat("Wh"),
+    Ore("Or"),
+    Sheep("Sh"),
+    Wood("Wd"),
+    Brick("Br")
+}
+
+// TODO find a way to make resourceYielded not nullable
+enum class HexTileType(val resourceYielded: ResourceType?) {
+    Field(ResourceType.Wheat),
+    Mountain(ResourceType.Ore),
+    Pasture(ResourceType.Sheep),
+    Forest(ResourceType.Wood),
+    Hill(ResourceType.Brick),
+    Desert(null)
```

Comment:
To avoid having a null here, would it make sense to add a `ResourceType.Void`?

Reply:
Yep - that works for me. It might take some refactoring but returning null is causing some tricky debugging situations.